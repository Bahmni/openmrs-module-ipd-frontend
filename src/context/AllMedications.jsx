import React, { useState } from "react";
import PropTypes from "prop-types";
import axios from "axios";
import { ALL_DRUG_ORDERS_URL } from "../constants";

export const AllMedicationsContext = React.createContext({
  getAllDrugOrders: () => {},
});

export const AllMedicationsContextProvider = (props) => {
  const getAllDrugOrders = async (visitUuid) => {
    setMedications((prevState) => ({ ...prevState, isLoading: true }));
    try {
      const response = await axios.get(ALL_DRUG_ORDERS_URL(visitUuid), {
        withCredentials: true,
      });
      if (response.status !== 200) throw new Error(response.statusText);
      setMedications((prevState) => ({
        ...prevState,
        isLoading: false,
        data: response.data,
      }));
    } catch (error) {
      setMedications((prevState) => ({
        ...prevState,
        isLoading: false,
        error,
      }));
    }
  };

  const initialState = {
    isLoading: false,
    data: null,
    error: null,
    getAllDrugOrders,
  };

  const [medications, setMedications] = useState(initialState);

  return (
    <AllMedicationsContext.Provider value={medications}>
      {props.children}
    </AllMedicationsContext.Provider>
  );
};

AllMedicationsContextProvider.propTypes = {
  children: PropTypes.children,
};
