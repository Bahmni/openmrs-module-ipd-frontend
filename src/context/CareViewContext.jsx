import React from "react";

export const CareViewContext = React.createContext();
