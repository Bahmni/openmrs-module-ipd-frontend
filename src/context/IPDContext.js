import React from "react";

export const IPDContext = React.createContext();