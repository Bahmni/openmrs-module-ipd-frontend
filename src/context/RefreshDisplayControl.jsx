import React from "react";

const RefreshDisplayControl = React.createContext({
  refreshDisplayControl: () => {},
});

export default RefreshDisplayControl;
