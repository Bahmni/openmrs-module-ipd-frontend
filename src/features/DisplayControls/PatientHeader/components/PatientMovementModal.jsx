import React, { useState, useContext, useEffect } from "react";
import { IPDContext } from "../../../../context/IPDContext";
import {
  fetchVisitSummary,
  updatePatientMovement,
  dischargePatient,
  undoDischargePatient,
} from "../utils/PatientMovementModalUtils";
import {
  searchConceptsByFSN,
  fetchVisitEncounterOrderTypes,
  getADTDashboardUrl,
} from "../../../../utils/CommonUtils";
import {
  FSN_DISPOSITION_VALUE,
  BY_FSN_VALUE,
  CUSTOM_OUTPUT_VALUE,
  FSN_ADT_NOTES_VALUE,
  BAHMNI_VALUE,
} from "../../../../constants";
import {
  Modal,
  TextArea,
  Dropdown,
  DropdownSkeleton,
} from "carbon-components-react";
import "../styles/PatientHeader.scss";
import PropTypes from 'prop-types';
import { drop } from "lodash";

const PatientMovementModal = (props) => {
  const { updatePatientMovementModal } = props;
  const { patient, visit, location, provider } = useContext(IPDContext);
  const [isLoading, setIsLoading] = useState(true);
  const [dropDown, setDropDown] = useState([]);
  const [visitSummary, setVisitSummary] = useState({});
  const [selectedDropDownItem, setSelectedDropDownItem] = useState({});
  const [currentItem, setCurrentItem] = useState({});
  const [adtNotes, setAdtNotes] = useState();
  const [adtNotesObj, setAdtNotesObj] = useState({});
  const [showAdtNotes, setShowAdtNotes] = useState(false);
  const [saveEnable, setSaveEnable] = useState(false);
  const [vistEncounterTypes, setVisitEncounterTypes] = useState({});
  const fetchDropDownValues = async () => {
    setIsLoading(true);
    const getDropDown = await searchConceptsByFSN(
      BY_FSN_VALUE,
      FSN_DISPOSITION_VALUE,
      CUSTOM_OUTPUT_VALUE
    );
    if (getDropDown.status === 200) {
      const items  = getDropDown.data.results[0].answers;
      var getVisitSummary = await fetchVisitSummary(visit);
      if (getVisitSummary.status === 200 && items.length > 0) {
        const visitSummaryData = getVisitSummary.data;
        setVisitSummary(visitSummaryData);
        var dropDownList = items.map((item) => {
          if (item.name) {
            if (
              visitSummaryData.admissionDetails === null &&
              visitSummaryData.dischargeDetails === null &&
              item.name.name === "Admit Patient"
            ) {
              return { label: item.name.display, value: item.name.display };
            } else if (
              visitSummaryData.admissionDetails !== null &&
              visitSummaryData.dischargeDetails !== null &&
              item.name.name === "Undo Discharge"
            ) {
              return { label: item.name.display, value: item.name.display };
            } else if (
              visitSummaryData.admissionDetails !== null &&
              visitSummaryData.dischargeDetails === null &&
              (item.name.name === "Discharge Patient" ||
                item.name.name === "Transfer Patient")
            ) {
              return { label: item.name.display, value: item.name.display };
            }
          }
        });
        dropDownList = dropDownList.filter(function( element ) {
          return element !== undefined;
       });
        setDropDown(dropDownList);
      }
      var getTypes = await fetchVisitEncounterOrderTypes();
      setVisitEncounterTypes(getTypes.data);
      if (getTypes.data) {
        setIsLoading(false);
      }
    }
  };

  useEffect(() => {
    fetchDropDownValues();
  }, []);

  const fetchAdtNotes = async () => {
    const getAdtNotes = await searchConceptsByFSN(
      BY_FSN_VALUE,
      FSN_ADT_NOTES_VALUE,
      BAHMNI_VALUE
    );
    if (getAdtNotes.status === 200) {
      setShowAdtNotes(true);
      setAdtNotesObj(getAdtNotes.data.results[0]);
    }
  };
  useEffect(() => {
    fetchAdtNotes();
  }, []);

  const createAdtNotesObj = () => {
    const notesObj = {
      concept: {
        uuid: adtNotesObj.uuid,
        name: adtNotesObj.name.name,
        dataType: adtNotesObj.datatype.name,
      },
      units: adtNotesObj.units,
      label: adtNotesObj.name.display,
      possibleAnswers: adtNotesObj.answers,
      groupMembers: adtNotesObj.setMembers,
      comment: null,
      isObservation: true,
      conceptUIConfig: [],
      value: adtNotes,
      autocompleteValue: adtNotes,
      conceptSetName: adtNotesObj.name.name,
    };
    return notesObj;
  };
  const createPayload = () => {
    const observations = adtNotes && showAdtNotes ? [createAdtNotesObj()] : [];
    const obj = {
      patientUuid: patient.uuid,
      encounterTypeUuid: selectedDropDownItem.encounterType,
      visitTypeUuid: vistEncounterTypes.visitTypes.IPD,
      observations: observations,
      locationUuid: location,
      providers: [provider],
    };
    return obj;
  };

  const handleOnSave = async () => {
    if (selectedDropDownItem) {
      setSaveEnable(false);
      if (selectedDropDownItem.name === "Admit Patient") {
        const response = await updatePatientMovement(createPayload());
        if (response.status === 200) {
          window.location.href = getADTDashboardUrl(
            patient.uuid,
            visit,
            response.data.encounterUuid
          );
        }
      } else if (selectedDropDownItem.name === "Discharge Patient") {
        const response = await dischargePatient(createPayload());
        if (response.status === 200) {
          updatePatientMovementModal(false);
        }
      } else if (selectedDropDownItem.name === "Undo Discharge") {
        const response = await undoDischargePatient(
          visitSummary.dischargeDetails.uuid,
          selectedDropDownItem.name
        );
        if (response.status === 200) {
          window.location.href = getADTDashboardUrl(
            patient.uuid,
            visit,
            visitSummary.dischargeDetails.uuid
          );
        }
      } else if (selectedDropDownItem.name === "Transfer Patient") {
        const response = await updatePatientMovement(createPayload());
        if (response.status === 200) {
          window.location.href = getADTDashboardUrl(
            patient.uuid,
            visit,
            response.data.encounterUuid
          );
        }
      }
    } else {
      setSaveEnable(true);
    }
  };

  const handleSelectOnChange = (e) => {
    var item = {};
    setCurrentItem(e);
    setSaveEnable(false);
    if (e.label === "Admit Patient") {
      item = {
        name: "Admit Patient",
        encounterType: vistEncounterTypes.encounterTypes.ADMISSION,
      };
      setShowAdtNotes(true);
    } else if (e.label === "Undo Discharge") {
      item = { name: "Undo Discharge", encounterType: null };
      setShowAdtNotes(false);
    } else if (e.label === "Discharge Patient") {
      item = {
        name: "Discharge Patient",
        encounterType: vistEncounterTypes.encounterTypes.DISCHARGE,
      };
      setShowAdtNotes(true);
    } else if (e.label === "Transfer Patient") {
      item = {
        name: "Transfer Patient",
        encounterType: vistEncounterTypes.encounterTypes.TRANSFER,
      };
      setShowAdtNotes(false);
    } else {
      setSaveEnable(true);
    }
    setSelectedDropDownItem(item);
  };

  const handleNotesOnChange = (e) => {
    setAdtNotes(e.target.value);
  };

  useEffect(() => {
    if (currentItem != {}) {
      setSaveEnable(false);
    }
    else {
      setSaveEnable(true);
    }
  },[currentItem]);
  
  return (
    <>
      {!isLoading && (
        <Modal
          open
          modalHeading="Patient Movement"
          primaryButtonText="Save"
          secondaryButtonText="Cancel"
          aria-label="Modal content"
          onRequestClose={() => updatePatientMovementModal(false)}
          onSecondarySubmit={() => updatePatientMovementModal(false)}
          onRequestSubmit={() => handleOnSave()}
          primaryButtonDisabled={saveEnable}
        >
          
          {dropDown == {} ? (
            <DropdownSkeleton />
          ) : (
            <Dropdown
              id="patient-movement-dropdown"
              placeholder={"Choose the patient movement"}
              titleText={""}
              onChange={(e) => handleSelectOnChange(e.selectedItem)}
              isRequired={true}
              width={"100%"}
              items={dropDown}
              itemToString={(item) => (item ? item.label : "")}
              selectedItem={currentItem}
            />
          )}
          &nbsp;&nbsp;
          {!showAdtNotes && (<div style={{height:"50px"}}></div>)}
          {showAdtNotes && (
            <TextArea
              data-modal-primary-focus
              id="ipd-adt-notes"
              placeholder={"Enter " + adtNotesObj.name.display}
              style={{
                marginBottom: "1rem",
              }}
              onChange={(e) => handleNotesOnChange(e)}
            />
          )}
        </Modal>
      )}
    </>
  );
};

export default PatientMovementModal;

PatientMovementModal.propTypes = {
  updatePatientMovementModal: PropTypes.func.isRequired
}
