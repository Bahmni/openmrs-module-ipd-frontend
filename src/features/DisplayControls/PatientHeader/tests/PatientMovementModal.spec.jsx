import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import PatientMovementModal from "../components/PatientMovementModal";
import {
  byFullySpecifiedNameForDispoistionMockData,
  byFullySpecifiedNameForAdtNotesMockData,
  visitSummaryToAdmitMockData,
  visitSummaryToDischargeOrTransferMockData,
  visitSummaryToUndoDischargeMockData,
  vistAndEncounterTypesMockData,
} from "./PatientMovementModalMockData";
import { IPDContext } from "../../../../context/IPDContext";
 
const mockSearchConceptsByFSN = jest.fn();
const mockFetchVisitSummary = jest.fn();
const mockFetchVisitEncounterOrderTypes = jest.fn();
const mockSearchConceptsByFSNForAdtNotes = jest.fn();
// const mockUpdatePatientMovement = jest.fn();
// const mockDischargePatient = jest.fn();
// const mockUndoDischargePatient = jest.fn();


const patientMock = "patientMockUuid";
const visitMock = "visitMockUuid";
const locationMock = "locationMockUuid";
const providerMock = "providerMockUuid";

jest.mock("../../../../utils/CommonUtils", () => {
  const originalCommonModule = jest.requireActual("../../../../utils/CommonUtils");
  return {
    ...originalCommonModule,
    searchConceptsByFSN: () => mockSearchConceptsByFSN("byFullySpecified", "Disposition", "custom:(uuid,name)"),
    fetchVisitEncounterOrderTypes: () => mockFetchVisitEncounterOrderTypes(),
    // searchConceptsByFSN: () => mockSearchConceptsByFSNForAdtNotes("byFullySpecified", "AdtNotes", "bahmni"),
    // updatePatientMovement: () => mockUpdatePatientMovement(),
    // dischargePatient: () => mockDischargePatient(),
    // undoDischargePatient: () => mockUndoDischargePatient(),
  };
});

jest.mock("../utils/PatientMovementModalUtils", () => {
  const originalModule = jest.requireActual("../utils/PatientMovementModalUtils");
  return {
    ...originalModule,
    fetchVisitSummary: () => mockFetchVisitSummary(visitMock),
    // updatePatientMovement: () => mockUpdatePatientMovement(),
    // dischargePatient: () => mockDischargePatient(),
    // undoDischargePatient: () => mockUndoDischargePatient(),
  };
});


describe("PatientMovementModal", () => {
  beforeEach(() => {
    mockSearchConceptsByFSN.mockResolvedValueOnce(byFullySpecifiedNameForDispoistionMockData).mockResolvedValueOnce(byFullySpecifiedNameForAdtNotesMockData);
    mockFetchVisitSummary.mockResolvedValue(visitSummaryToAdmitMockData);
    mockFetchVisitEncounterOrderTypes.mockResolvedValue(vistAndEncounterTypesMockData);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it("should display patient movement modal", async () => {
    render(<IPDContext.Provider value={{patient: patientMock, visit: visitMock, location: locationMock, provider: providerMock}}>
      <PatientMovementModal updatePatientMovementModal={()=> jest.fn()}/></IPDContext.Provider>);
    // render(<PatientMovementModal updatePatientMovementModal={()=> jest.fn()}/>);

    await waitFor(() => expect(screen.getByText("Patient Movement")).toBeTruthy());
  });

  it("should display patient movement modal with a dropdown", async () => {
    render(<IPDContext.Provider value={{patient: patientMock, visit: visitMock, location: locationMock, provider: providerMock}}>
      <PatientMovementModal updatePatientMovementModal={()=> jest.fn()}/></IPDContext.Provider>);
    
    await waitFor(() => {
      expect(screen.getByText("Select")).toBeTruthy();
      expect(screen.getByText("Choose the patient movement")).toBeTruthy();
     
    });
  }); 

  it("should display patient movement modal with a text area", async () => {
    render(<IPDContext.Provider value={{patient: patientMock, visit: visitMock, location: locationMock, provider: providerMock}}>
      <PatientMovementModal updatePatientMovementModal={()=> jest.fn()}/></IPDContext.Provider>);
    
    await waitFor(() => {
      expect(screen.getByPlaceholderText("Enter Adt Notes")).toBeTruthy();
    });
  }); 

  it("should close the patient modal when close icon is clicked", async () => {
    const closeMethod = jest.fn();
    render(<IPDContext.Provider value={{patient: patientMock, visit: visitMock, location: locationMock, provider: providerMock}}>
      <PatientMovementModal updatePatientMovementModal={closeMethod}/></IPDContext.Provider>);

    await waitFor(() => {
      expect(screen.getByText("Patient Movement")).toBeTruthy();
    });
    const closeButton = screen.getByRole('button', {name: "close"});
    expect(closeButton).toBeTruthy()
    fireEvent.click(closeButton);
    expect(closeMethod).toBeCalledTimes(1);
  });

  it("should close the patient modal when cancel button is clicked", async () => {
    const cancelMethod = jest.fn();
    render(<IPDContext.Provider value={{patient: patientMock, visit: visitMock, location: locationMock, provider: providerMock}}>
      <PatientMovementModal updatePatientMovementModal={cancelMethod}/></IPDContext.Provider>);

    await waitFor(() => {
      expect(screen.getByText("Patient Movement")).toBeTruthy();
    });
    const cancelButton = screen.getByText("Cancel");
    expect(cancelButton).toBeTruthy()
    fireEvent.click(cancelButton);
    expect(cancelMethod).toBeCalledTimes(1);
  });

  it('should display patient movement modal with a admit dropdown', async () => {

    render(<IPDContext.Provider value={{patient: patientMock, visit: visitMock, location: locationMock, provider: providerMock}}>
      <PatientMovementModal updatePatientMovementModal={()=> jest.fn()}/></IPDContext.Provider>);
    
    await waitFor(() => {
      expect(screen.getByText("Patient Movement")).toBeTruthy();
    });
    expect(screen.getByText("Admit Patient")).toBeTruthy();
    
  });

  it('should save patient movement modal for admit patient', async () => {
    const { container } = render(<IPDContext.Provider value={{patient: patientMock, visit: visitMock, location: locationMock, provider: providerMock}}>
      <PatientMovementModal updatePatientMovementModal={()=> jest.fn()}/></IPDContext.Provider>);
       await waitFor(() => {
        expect(screen.getByText("Patient Movement")).toBeTruthy();
      });
      
    const saveButton = screen.getByText("Save");
    expect(saveButton.disabled).toEqual(true);

    
    // screen.debug();
    const chooseDropdown = screen.getByText("Choose the patient movement")
    fireEvent.click(chooseDropdown);
    expect(screen.getByText("Admit Patient")).toBeTruthy();
    fireEvent.click(screen.getByText("Admit Patient"));
    expect(saveButton.disabled).toEqual(false);
    
    // const addNotes = screen.getByRole('textbox');
    // fireEvent.change(addNotes, { target: { value: "Adt Notes" } });
    // expect(addNotes.value).toEqual("Adt Notes");
    
    // expect(saveButton.disabled).toEqual(false);
    //fireEvent.click(saveButton);
    // check whether that is updatePatient method is called
  });

  it('should display patient movement modal with a discharge and transfer dropdown', async () => {
    mockFetchVisitSummary.mockResolvedValue(visitSummaryToDischargeOrTransferMockData);
    render(<IPDContext.Provider value={{patient: patientMock, visit: visitMock, location: locationMock, provider: providerMock}}>
      <PatientMovementModal updatePatientMovementModal={()=> jest.fn()}/></IPDContext.Provider>);
    
    await waitFor(() => {
      expect(screen.getByText("Patient Movement")).toBeTruthy();
    });
    expect(screen.getByText("Discharge Patient")).toBeTruthy();
    expect(screen.getByText("Transfer Patient")).toBeTruthy();
  });

  // it('should save patient movement modal for discharge patient', async () => {
  
  // });

  // it('should save patient movement modal for transfer patient', async () => {
  
  // });

  it('should display patient movement modal with a undo discharge dropdown', async () => {
    mockFetchVisitSummary.mockResolvedValue(visitSummaryToUndoDischargeMockData);
    render(<IPDContext.Provider value={{patient: patientMock, visit: visitMock, location: locationMock, provider: providerMock}}>
      <PatientMovementModal updatePatientMovementModal={()=> jest.fn()}/></IPDContext.Provider>);
    
    await waitFor(() => {
      expect(screen.getByText("Patient Movement")).toBeTruthy();
    });
    expect(screen.getByText("Undo Discharge")).toBeTruthy();
  });

  // it('should save patient movement modal for undo discharge patient', async () => {
  
  // });

   

  // it("should render without crashing", () => {
  //   render(<PatientHeader patientId="123" />);
  // });

  // it("should call fetchPatientInfo on mount", () => {
  //   render(<PatientHeader patientId="123" />);
  //   expect(mockFetchPatientProfile).toHaveBeenCalledWith("123");
  // });

  // it("should display loading skeleton while fetching data", () => {
  //   render(<PatientHeader patientId="123" />);
  //   expect(screen.getByTestId("header-loading")).toBeTruthy();
  // });

  // it("should display patient details after data is fetched", async () => {
  //   render(<PatientHeader patientId="123" />);
  //   await waitFor(() => expect(screen.getByText("John Doe")).toBeTruthy());
  //   expect(screen.getByText(/30 Years/i)).toBeTruthy();
  //   expect(screen.getByText("01/01/1991")).toBeTruthy();
  //   expect(screen.getByText(/12345/i)).toBeTruthy();
  // });

  // it("should display all details of the patient", async () => {
  //   const { container } = render(<PatientHeader patientId="123" />);
  //   await waitFor(() => expect(screen.getByText("John Doe")).toBeTruthy());
  //   const showDetailsButton = screen.getByText("Show Details");
  //   fireEvent.click(showDetailsButton);

  //   await waitFor(() => {
  //     expect(screen.getByText(/Country : Ethiopia/i)).toBeTruthy();
  //     expect(container).toMatchSnapshot();
  //   });
  // });

  // it("should display patient movement item on click of overflow menu icon", async () => {
  //   const { container } = render(<PatientHeader patientId="123" />);
  //   await waitFor(() => expect(screen.getByText("John Doe")).toBeTruthy());
  //   screen.getByTestId("overflow-menu").click();
  //   expect(screen.getByText("Patient Movement")).toBeTruthy();
  //   expect(container).toMatchSnapshot();
  // });
});
