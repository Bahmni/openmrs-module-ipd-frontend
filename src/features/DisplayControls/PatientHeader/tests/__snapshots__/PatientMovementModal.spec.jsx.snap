// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientHeader should display patient movement item on click of overflow menu icon 1`] = `
<div>
  <div
    class="bx--tile patient-header"
  >
    <div
      class="header-title-grid bx--grid"
    >
      <div
        class="patient-image-and-details bx--row"
      >
        <div
          class="patient-image"
        />
        <div
          class="bx--col"
        >
          <div
            class="header-title-row bx--row"
          >
            <div
              class="patient-name-and-navigations"
            >
              <h1
                class="patient-name"
              >
                John Doe
              </h1>
              <a
                class="bx--link"
              >
                Patient Dashboard
              </a>
              <a
                class="bx--link"
              >
                Discharge Summary
              </a>
            </div>
            <button
              aria-expanded="true"
              aria-haspopup="true"
              aria-label="Open and close list of options"
              class="bx--overflow-menu bx--overflow-menu--open bx--overflow-menu--md"
              data-testid="overflow-menu"
              type="button"
            >
              <svg
                aria-label="Open and close list of options"
                class="bx--overflow-menu__icon"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                role="img"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="16"
                  cy="8"
                  r="2"
                />
                <circle
                  cx="16"
                  cy="16"
                  r="2"
                />
                <circle
                  cx="16"
                  cy="24"
                  r="2"
                />
                <title>
                  Open and close list of options
                </title>
              </svg>
            </button>
          </div>
          <div
            class="bx--row"
          >
            <div
              class="other-info"
            >
              <div
                class="patient-basic-info"
              >
                <h3
                  class="patient-info"
                >
                  Male
                </h3>
                <h3
                  class="patient-info"
                >
                  30
                   
                  Years
                </h3>
                <h3
                  class="patient-info"
                >
                  01/01/1991
                </h3>
                <h3
                  class="patient-info"
                >
                  12345
                </h3>
              </div>
              <a
                class="bx--link show-more bx--link--sm"
                kind="tertiary"
              >
                Show Details
                 
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 22L6 12 7.4 10.6 16 19.2 24.6 10.6 26 12z"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
